js/data/bucket/symbol_bucket.js

```diff
@@ -339,23 +339,20 @@ SymbolBucket.prototype.placeFeatures = function(collisionTile, showCollisionBoxe
 
     this.createArrays();
 
-    var elementGroups = this.elementGroups = {
-        glyph: [],
-        icon: [],
+    var layout = this.layer.layout;
+
+    this.metadata = {
         sdfIcons: this.sdfIcons,
-        iconsNeedLinear: this.iconsNeedLinear
+        iconsNeedLinear: this.iconsNeedLinear,
+        adjustedTextSize: this.adjustedTextSize,
+        adjustedIconSize: this.adjustedIconSize,
+        // Transfer the name of the fonstack back to the main thread along with the buffers.
+        // The draw function needs to know which fonstack's glyph atlas to bind when rendering.
+        fontstack: layout['text-font'].join(',')
```

Comment:
The difference between `this.metadata[key]` properties and `this[key]` properties is that `this.metadata[key]` properties are specific to a particular subclass of `Bucket`? That strikes me as unconventional software architecture. 

> This is good because the serialization code is in Bucket and these individual properties are specific to SymbolBucket

I would handle this by creating a `SymbolBucket#serialize` method that calls `Bucket.prototype.serialize`